tests/python_package_test/test_basic.py

```diff
@@ -913,6 +913,17 @@ def test_feature_names_are_set_correctly_when_no_feature_names_passed_into_Datas
     assert ds.construct().feature_name == ["Column_0", "Column_1", "Column_2"]
 
 
+@pytest.mark.parametrize("max_depth", [4, 8, 12, 16])
```

Comment:
What is the purpose of testing across these different values of `max_depth` for this? Could you give a concrete example of a way that the implementation might fail for one of these values but not the other?